### PR TITLE
[soft_165] - Ads1015 smoke test

### PR DESCRIPTION
**Pull request by tristanle22**
Smoke test for ads1015:

- Take a reading periodically and log the result
- Configurable Items: wait time, I2C port, channels to be tested, conversion enable

### CODE REVIEW

File: projects/smoke_ads1015/src/main.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  } else {
+    for (size_t i = 0; i < SIZEOF_ARRAY(test_channels) && status == STATUS_CODE_OK; i++) {
+      status = ads1015_read_converted(storage, test_channels[i], &reading);
+      if (status == STATUS_CODE_OK)
+        LOG_DEBUG("Channel: %d; Converted value: %d mVolt\n", test_channels[i], reading);
+      else
+        LOG_DEBUG("Fail to take converted reading\n");
+    }
+  }
+  soft_timer_start_millis(SMOKE_READING_WAIT_MS, prv_read_and_log, &s_storage, NULL);
```
[the remark this answers is not in the document]
Comment (tristanle22) [reply projects/smoke_ads1015/src/main.c:48]: hm it's not tho. Should I declare storage as a pointer?

[the remark this answers is not in the document]
Comment (g-jessmuir) [reply projects/smoke_ads1015/src/main.c:49]: `Ads1015Storage *storage` is a pointer, while `s_storage` is not. I think it's reasonably descriptive whichever one you choose, up to you, resolve if you want to just leave as is

**Comment by g-jessmuir** [comment]
Please address or resolve the remaining two comments then we can merge, thanks!

File: projects/smoke_ads1015/src/main.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (tristanle22) [reply projects/smoke_ads1015/src/main.c:49]: Ohh I see what you mean, I thought we're talking about the soft_timer call in main(). I'll fix it, looks nicer anw. Thanks for pointing that out!

**Pull request merged**